WalletWasabi.Fluent/Models/UI/UIContext.cs

```diff
@@ -0,0 +1,38 @@
+using Avalonia;
+using Avalonia.Input.Platform;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+
+namespace WalletWasabi.Fluent.Models.UI;
+
+public class UIContext
+{
+	private INavigate? _navigate;
+
+	public UIContext(IQrCodeGenerator qrCodeGenerator, IClipboard clipboard)
+	{
+		QrCodeGenerator = qrCodeGenerator;
+		Clipboard = clipboard;
+	}
+
+	public IClipboard Clipboard { get; }
+	public IQrCodeGenerator QrCodeGenerator { get; }
+
+	public static UIContext Default => new(new QrGenerator(), Application.Current?.Clipboard);
+
+	public void RegisterNavigation(INavigate navigate)
```

Comment:
Wouldn't it be better to initialize _navigate in the constructor like the other services?

Reply:
You can't do that because `INavigate` is implemented by `NavigationState` which requires ViewModels which are children of `MainViewModel`. A larger refactoring is needed to achieve that.